src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -320,6 +321,250 @@ public async Task binding_in_variable_value_is_valid()
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "custom_date" : "{{$datetime 'yyyy-MM-dd'}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var currentDate = DateTime.UtcNow.ToString("yyyy-MM-dd");
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        bodyAsString.Should().BeExceptingWhitespace($$"""
+            {
+                "custom_date" : "{{currentDate}}"
+            }
+            """);
+
+    }
+
+    [Fact]
+    public async Task can_bind_local_datetime_with_parameters()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "local_custom_date" : "{{$localDatetime 'yyyy-MM-dd'}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var currentDate = DateTime.UtcNow.ToString("yyyy-MM-dd");
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        bodyAsString.Should().BeExceptingWhitespace($$"""
+            {
+                "local_custom_date" : "{{currentDate}}"
+            }
+            """);
+    }
+
+    [Fact]
+    public async Task several_system_variables_in_single_request()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "request_id": "{{$guid}}",
+                "updated_at": "{{$timestamp}}",
+                "created_at": "{{$timestamp -1 d}}",
+                "review_count": "{{$randomInt 5 200}}",
+                "custom_date": "{{$datetime 'yyyy-MM-dd'}}",
+                "local_custom_date": "{{$localDatetime 'yyyy-MM-dd'}}"
```

Comment:
Consider sprinkling some of these in other parts of request (e.g. URL, headers etc.)